File: internal/model/task.go
package model

import (
	"time"
)

type Priority string

const (
	PriorityHigh   Priority = "High"
	PriorityMedium Priority = "Medium"
	PriorityLow    Priority = "Low"
)

func (p Priority) IsValid() bool {
	switch p {
	case PriorityHigh, PriorityMedium, PriorityLow:
		return true
	}
	return false
}

type List struct {
	ID              string    `json:"id"`
	Name            string    `json:"name"`
	Color           string    `json:"color"`
	CreatedAt       time.Time `json:"created_at"`
	TotalCount      int       `json:"total_count"`
	IncompleteCount int       `json:"incomplete_count"`
}

type Task struct {
	ID          string     `json:"id"`
	ListID      *string    `json:"list_id"`
	Title       string     `json:"title"`
	Description *string    `json:"description,omitempty"`
	Priority    *Priority  `json:"priority,omitempty"`
	Deadline    *time.Time `json:"deadline,omitempty"`
	Status      bool       `json:"status"`
	CreatedAt   time.Time  `json:"created_at"`
	UpdatedAt   time.Time  `json:"updated_at"`
}

File: web/src/types/task.ts
export type Priority = "High" | "Medium" | "Low";

export interface List {
  id: string;
  name: string;
  color: string;
  created_at: string;
  total_count: number;
  incomplete_count: number;
}


export interface Task {
  id: string;
  list_id: string | null;
  title: string;
  description?: string;
  priority?: Priority;
  deadline?: string;
  status: boolean;
  completed_at?: string;
  created_at: string;
  updated_at: string;
}

export interface CreateTaskRequest {
  list_id?: string;
  title: string;
  description?: string;
  priority?: Priority;
  deadline?: string;
}

export interface GetTasksResponse {
  data: Task[] | null;
  total: number;
  page: number;
  page_size: number;
}


